raiden-ts/src/raiden.ts

```diff
@@ -1043,14 +1044,21 @@ export class Raiden {
     // Check whether we are on a test network
     assert(this.deps.network.chainId !== 1, ErrorCodes.RDN_MINT_MAINNET, this.log.info);
 
-    const { signer } = chooseOnchainAccount(this.deps, subkey ?? this.config.subkey);
+    const { signer, address } = chooseOnchainAccount(this.deps, subkey ?? this.config.subkey);
     // Mint token
     const customTokenContract = CustomTokenFactory.connect(token, signer);
 
+    const beneficiary = to ?? address;
+    assert(
+      Address.is(beneficiary),
+      [ErrorCodes.DTA_INVALID_ADDRESS, { beneficiary }],
+      this.log.info,
+    );
+
     const receipt = await callAndWaitMined(
       customTokenContract,
-      'mint',
-      [decode(UInt(32), amount)],
+      'mintFor',
```

Comment:
Hmm the docs say that the token contract can either define `mint(address, uint256)`, `mintFor(uint256, address)` or `increaseSupply(uint256, address)`. The last seem to be not relevant. But from the first two: what is the difference in behavior except the signature? Is there definition exclusive? Should we maybe check for both? Or is `mintFor` the only one that does what we want now?

Reply:
We don't support all these interfaces. If you look into the source of the CustomToken contract, you'll see it supports just 2 minting methods: `mint(uint256)`, which mints for self, and `mintFor(uint256, address)`, which allows minting for a 3rd party. This is a `_testing` function, used only on our testnets, and these are always deployed with this contract. It'd be hard for the SDK to support multiple contracts interfaces, so we went with what is actually used.